### DISCUSSION TITLE
Access notes for virtual fretboard

### DISCUSSION THREAD

**Discussion by macca007x** [CoderLine/alphaTab#1685] [Q&A] [answered]
Is there an easier way to access the notes + effects present on them?

Thanks.

**Comment by macca007x** [comment, marked as answer]
This works

```
api.playedBeatChanged.on((beat) => {
      console.log(beat);
    });
```

**Reply by Danielku15** [reply]
Exactly. More info around your need of building a fretboard:

* https://www.alphatab.net/docs/reference/api/activebeatschanged/
* https://www.alphatab.net/docs/reference/api/playedbeatchanged/
* https://www.alphatab.net/docs/reference/score
